app/services/db/monthly_sp_auth_count/unique_monthly_auth_counts_by_partner.rb

```diff
@@ -0,0 +1,151 @@
+# frozen_string_literal: true
+
+module Db
+  module MonthlySpAuthCount
+    module UniqueMonthlyAuthCountsByPartner
+      extend Reports::QueryHelpers
+
+      module_function
+
+      # @param [String] key label for billing (Partner requesting agency)
+      # @param [Array<String>] issuers issuers for the iaa
+      # @param [Date] start_date iaa start date
+      # @param [Date] end_date iaa end date
+      # @return [PG::Result, Array]
+      def call(key:, issuers:, start_date:, end_date:)
+        date_range = start_date...end_date
+
+        return [] if !date_range || issuers.blank?
+
+        # Query a month at a time, to keep query time/result size fairly reasonable
+        # The results are rows with [user_id, ial, auth_count, year_month]
+        months = Reports::MonthHelper.months(date_range)
+        queries = build_queries(issuers: issuers, months: months)
+
+        year_month_to_users_to_profile_age = Hash.new do |ym_h, ym_k|
+          ym_h[ym_k] = Hash.new { |user_h, user_k| }
```

Comment:
the last block here is empty. either we meant to have `Hash.new` (no block) or the goal is to initialize the user hash:
```suggestion
          ym_h[ym_k] = Hash.new { |user_h, user_k| user_h[user_k] = Hash.new }
```

Reply:
Do we need a hash for profile_age?  I think we were mistaken when we built that third level - Every `year_month` will have multiple `users`, and every `user` will have only one `profile_age` coming from the query.  Correct?

Reply:
Resolved in person